weblate/screenshots/models.py

```diff
@@ -76,3 +86,29 @@ def change_screenshot_assignment(sender, instance, action, **kwargs):
         name="UnusedScreenshot"
     ).exists():
         component_alerts.delay([instance.pk])
+
+
+@receiver(vcs_post_update)
+def update_screenshots_from_repo(sender, component, previous_head: str, **kwargs):
+    screenshots = Screenshot.objects.filter(translation__component=component)
+    repo = component.repository
+
+    # Todo: Add suport for all supported VCSes
+    changed_files = repo.list_changed_files("refspec")
+
+    for s_shot in screenshots:
+        f_name = s_shot.repository_filename
+        if f_name in changed_files and component.changed_screenshot_files(f_name):
+            current_screenshot_object = s_shot.image
+            new_screenshot_image = repo.get_file(f_name, previous_head)
+
+            # Validate Image
+            try:
+                validate_bitmap(new_screenshot_image)
+            except ValidationError:
+                report_error(cause="Failed to validate image.")
```

Comment:
Failed validation should stop further processing:

```suggestion
                report_error(cause="Failed to validate image from repository")
                continue
```